### ISSUE TITLE
restore_command is removed from recovery.conf after restore completes

### ISSUE THREAD

**Issue by jonjomckay** [sorintlab/stolon#537]
### Submission type

 - Bug report

#### Stolon version
v0.12.0
### Expected behaviour you didn't see
I expected the given `restoreCommand` value to be present in the `recovery.conf` file after restore completes and recovery starts, but it's missing.
### Unexpected behaviour you saw
No `restore_command` setting was present in `recovery.conf` after the initial data restore completes, so recovery fails with the error: `FATAL:  could not receive data from WAL stream: ERROR:  requested WAL segment 0000000100000CC000000004 has already been removed`.

The error is correct, as the WAL segment no longer exists on the instance we're replicating from, but it should be falling back to load from the `restore_command` (which would succeed), as we ship our WAL segments to external storage rather than have them reside on the master (we use WAL-E, FWIW). We already have this functionality working for the past 3 years on a non-Stolon cluster.
### Steps to reproduce the problem

1. Initialize the cluster with this spec:

    ```
    {
      "initMode": "pitr",
      "pitrConfig": {
        "dataRestoreCommand": "WALE_S3_PREFIX=s3://bucket-name AWS_REGION=us-east-1 /usr/local/bin/wal-e backup-fetch %d LATEST",
        "archiveRecoverySettings": {
          "restoreCommand": "WALE_S3_PREFIX=s3://bucket-name AWS_REGION=us-east-1 /usr/local/bin/wal-e wal-fetch \"%f\" \"%p\""
        }
      },
      "role": "standby",
      "standbySettings": {
        "primaryConnInfo": "host=master-instance port=5432 user=replicator password=password sslmode=prefer"
      },
      "usePgrewind": true
    }
    ```
2. Wait for the initial data restore to complete successfully and for PostgreSQL to restart
3. The following error occurs:

    ```
    LOG:  consistent recovery state reached at CC0/308ED20
    LOG:  database system is ready to accept read only connections
    2018-07-28T13:52:51.944Z    INFO    postgresql/postgresql.go:351    stopping database
    LOG:  received fast shutdown request
    LOG:  aborting any active transactions
    LOG:  shutting down
    LOG:  database system is shut down
    waiting for server to shut down.... done
    server stopped
    2018-07-28T13:52:52.981Z    INFO    cmd/keeper.go:1456  already standby
    2018-07-28T13:52:52.993Z    INFO    postgresql/postgresql.go:287    starting database
    LOG:  could not open usermap file "/stolon-data/postgres/pg_ident.conf": No such file or directory
    LOG:  database system was shut down in recovery at 2018-07-28 13:52:52 GMT
    LOG:  entering standby mode
    LOG:  redo starts at CBF/FE15F318
    LOG:  consistent recovery state reached at CC0/308ED20
    LOG:  database system is ready to accept read only connections
    2018-07-28T13:52:53.209Z    INFO    cmd/keeper.go:1585  postgres parameters not changed
    2018-07-28T13:52:53.209Z    INFO    cmd/keeper.go:1596  postgres hba entries not changed
    LOG:  started streaming WAL from primary at CC0/4000000 on timeline 1
    FATAL:  could not receive data from WAL stream: ERROR:  requested WAL segment 0000000100000CC000000004 has already been removed
    ```

**Comment by jonjomckay** [comment]
@sgotti that's right, but we should be able to restore from the WAL segments we have stored before switching to streaming replication (as we do already on non-Stolon slaves). I can confirm the requested segment `0000000100000CC000000004` does exist in our storage so it should have been restored. In theory, we should be able to continuously pull WALs from storage and not talk to the master at all.

Do you have any tips on using WAL-E with Stolon, where I could be missing something? I've followed [this one](https://github.com/sorintlab/stolon/blob/master/doc/pitr_wal-e.md) already. Thanks!

**Comment by jonjomckay** [comment]
@sgotti me again! FWIW, I tried manually adding `restore_command = 'WALE_S3_PREFIX=s3://bucket-name AWS_REGION=us-east-1 /usr/local/bin/wal-e wal-fetch "%f" "%p"'` directly into my container's `/stolon-data/postgres/recovery.conf`, ran a `pg_ctl restart` and the restore completed successfully. If I restart the pod though, it looks like Stolon overwrites the `recovery.conf` with its own version, breaking the process again :(

**Comment by sgotti** [comment]
> In theory, we should be able to continuously pull WALs from storage and not talk to the master at all.

That's currently not supported, once done the PITR phase the standby cluster phase wants to fetch wals from the streaming protocol. It could be enhanced to fetch wals using the restore_command (with many downsides like not having the standby cluster lagging behind a lot). 

Please open a new issue as an RFE or create a PR implementing that feature.

> FWIW, I tried manually adding restore_command = 'WALE_S3_PREFIX=s3://bucket-name AWS_REGION=us-east-1 /usr/local/bin/wal-e wal-fetch "%f" "%p"' directly into my container's /stolon-data/postgres/recovery.conf, ran a pg_ctl restart and the restore completed successfully. If I restart the pod though, it looks like Stolon overwrites the recovery.conf with its own version, breaking the process again :(

That's wanted, stolon manages the recovery.conf file so you cannot overwrite it.

**Comment by sgotti** [comment]
Fixed by #543
